ui/src/message_popup/message_popup_page.jsx

```diff
@@ -133,33 +181,65 @@ export default React.createClass({
       questions,
       questionsAnswered,
       shouldShowStudentCards,
-      helpType
+      shouldShowSummary,
+      helpType,
+      sessionLength
     } = this.state;
     if (!hasStarted) return this.renderInstructions();
-    if (questionsAnswered >= questions.length) return this.renderDone();
+    if (questionsAnswered >= sessionLength) return this.renderDone();
 
     const question = this.state.questions[questionsAnswered];
     return (
       <div style={styles.container}>
+        <LinearProgress mode="determinate" value={questionsAnswered} max={sessionLength} />
         <VelocityTransitionGroup enter={{animation: "slideDown"}} leave={{animation: "slideUp"}} runOnMount={true}>
           <PopupQuestion
             key={JSON.stringify(question)}
             question={question}
             shouldShowStudentCard={shouldShowStudentCards}
+            shouldShowSummary={shouldShowSummary}
             helpType={helpType}
             limitMs={90000}
             onLog={this.onLog}
-            onDone={this.onQuestionDone}/>
+            onDone={this.onQuestionDone}
+            addResponseTime={this.addResponseTime}
+            isLastQuestion={questionsAnswered+1===sessionLength ? true : false}/>
         </VelocityTransitionGroup>
+        <Snackbar
+          open={this.state.toastRevision}
+          message="Response recorded for feedback"
+          autoHideDuration={2000}
+          onRequestClose={this.removeToast}
+        />
       </div>
     );
   },
 
   renderDone() {
+    const mean = Math.round(_.mean(this.state.responseTimes)/10)/100;
```

Comment:
Nice!  There's a bit of computation in here, I'd suggest splitting this bit out to its own component that owns computing and showing this summary.

Also, can you revise this to use full words over abbreviations like `avg.` and `#`?